### ISSUE TITLE
False positive: Snippet prop typing error with @render

### ISSUE THREAD

**Issue by janosh** [pheuter/svelte-check-rs#94]
When passing a snippet as a children prop to a component, svelte-check-rs reports a type error about \`{@render ...} must be called with a Snippet\`.

## Reproduction

```svelte
<script lang="ts">
  import Select from 'svelte-multiselect'
  let value = $state('option1')
</script>

<Select options={['option1', 'option2']} bind:value>
  {#snippet children({ option })}
    <span>{option}</span>
  {/snippet}
</Select>
```

## Error

```
Error: Type '(this: void, { option }: { option: Option; idx: number; }) => { '{@render ...} must be called with a Snippet': "import type { Snippet } from 'svelte'"; } & unique symbol' is not assignable to type 'Snippet<[{ option: Option; idx: number; }]>'. (ts(TS2322))
```

The error message itself is unusual - it suggests the return type includes a string literal about how to use snippets.

svelte-check-rs: 0.7.3

**Comment by pheuter** [comment]
Thanks for the report! I attempted to reproduce this issue but was **unable to replicate the error** with the current versions.

**Test setup:**
- svelte-check-rs 0.7.3
- svelte 5.45.6
- svelte-check 4.3.4 (for comparison)
- svelte-multiselect 11.6.0

I created a reproduction with:
1. A generic component accepting a typed snippet prop (`Snippet<[{ option: Option; idx: number }]>`)
2. A page using it with an inline `{#snippet children({ option })}` block

Both `svelte-check-rs` and the original `svelte-check` report no errors.

Could you provide:
- Your TypeScript version
- Your `tsconfig.json` contents
- A minimal reproducible example (ideally a GitHub repo or tarball)
- Whether this still occurs with the latest versions

This would help narrow down whether it's environment-specific or related to particular type combinations in `svelte-multiselect`.
